src/coreclr/vm/prestub.cpp

```diff
@@ -1032,6 +1042,629 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, COR_ILMETHOD_
     return pCode;
 }
 
+bool MethodDesc::TryGenerateTransientILImplementation(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    if (TryGenerateAsyncThunk(resolver, methodILDecoder))
+    {
+        return true;
+    }
+
+    if (TryGenerateUnsafeAccessor(resolver, methodILDecoder))
+    {
+        return true;
+    }
+
+    return false;
+}
+
+bool MethodDesc::TryGenerateAsyncThunk(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
```

Comment:
Can the generation of all async helpers be in a separate file for readability? TryGenerateUnsafeAccessor was moved to a separate file for the same reason.

Reply:
Moved async helpers to a different file. (managed helpers too)